modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,140 @@
+= Kafka - Upgrade and Migrate - Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../../assets/images/
+
+Upgrade Anypoint Connector for Kafka (Kafka Connector) to version 4.0.0
+
+== Supported Upgrade Paths
+
+[%header,cols="50a,50a"]
+|===
+|From Version | To Version
+|3.2.x |4.0.0
+|===
+
+== Changes in This Release
+
+This release contains the following changes:
+
+* New operations, with new parameters. (see in the: New Operations, Parameters table)
+
+* The Message Listener replaces the Message Consumer Source, it has new parameters are shown in the New Source Table. (see in the: Changed Source table)
+
+* New Source was added (see in the: New Source table)
+
+* Added different Ack modes in the Consumer Configuration AUTO: Mule ACKs the message only if the flow is finished successfully; MANUAL: The user must do the ack manually within the flow; DUPS_OK: Same as the AUTO mode, but the commit is made asynchronously, which can lead to duplicate records; IMMEDIATE: Mule automatically ACKs the message upon reception.
+
+* Added Transactional SASL connection, this connection type is the same as the Producer SASL connection, but with the difference that it’s not a cached connection and is transactional.
+
+* Added Transactional plaintext connection, this connection type is the same as the Producer plaintext connection, but with the difference that it’s not a cached connection and is transactional.
+
+== Changed Source, Old version, Parameters, and Return Types
+
+[%header%autowidth.spread]
+|===
+|Kafka Source | Old version | Parameters
+
+| Message listener | Message consumer | ** Poll timeout; ** Poll timeout time unit; ** Acknowledgement mode; ** Amount of parallel consumers
+
+|===
+
+== New Operations, Parameters, and Return Types
+
+[%header%autowidth.spread]
+|===
+|Kafka Operation | Description | Parameters
+
+| Commit
+a| Commits the offsets associated to a message or batch of messages consumed in a Message Listener. |
+** Consumer commit key; ** (The consumer commit key to use to commit the message.)
```

Comment:
Add "The commitKey is inserted by the MessageListeners and Consume operation as a parameter of the message"